packages/core/src/di/provider_collection.ts

```diff
@@ -251,12 +263,12 @@ export function walkProviderTree(
     }
 
     // Next, include providers listed on the definition itself.
-    const defProviders = injDef.providers;
+    const defProviders = injDef.providers as Array<SingleProvider|InternalEnvironmentProviders>;
     if (defProviders != null && !isDuplicate) {
       const injectorType = container as InjectorType<any>;
-      deepForEach(defProviders, provider => {
-        ngDevMode && validateProvider(provider, defProviders as SingleProvider[], injectorType);
-        providersOut.push(provider);
+      deepForEachProvider(defProviders, provider => {
```

Comment:
Just curious if we can type it here (vs adding `as`)?
```suggestion
      deepForEachProvider(defProviders, (provider: SingleProvider) => {
```